internal/verifier/verifier.go

```diff
@@ -122,3 +128,47 @@ func (verifier *AmberProvenanceMetadataVerifier) Verify(provenanceFilePath strin
 
 	return nil
 }
+
+// Internal intermediate representation of Provenances for verification.
+// We use the ProvenanceIR to
+// (1) map different provenance formats, and
+// (2) hold reference values.
+// To be usable with different provenance formats, we allow fields to be empty ([]) and to hold several reference values.
+type ProvenanceIR struct {
+	BinarySHA256Digest []string
+}
+
+func (p *ProvenanceIR) FromSLSA(provenance *slsa.ValidatedProvenance) {
```

Comment:
Hmmm... I have been thinking, but I don't see a strong reason for them. 

The mapping should be obvious from the code. 

We could use unit tests to document our assumptions, e.g., that "*amber.ValidatedProvenance contains a SHA256 hash of a single subject", but that should actually be implied by the validation.

Do you see a good case?

Reply:
It will make refactoring easier. When the logic gets more complicated, and if we decide to make changes to the mapping, having tests will make it easier. But we can also add tests once there are more fields in ProvenanceIR.